third_party/3/aiofiles/base.pyi

```diff
@@ -1,17 +1,18 @@
 from types import CodeType, FrameType, TracebackType, coroutine
 from typing import Any, Coroutine, Generator, Generic, Iterator, Optional, Type, TypeVar, Union
 
+_T = TypeVar("_T")
 _T_co = TypeVar("_T_co", covariant=True)
 _V_co = TypeVar("_V_co", covariant=True)
 _T_contra = TypeVar("_T_contra", contravariant=True)
 
-class AsyncBase:
+class AsyncBase(Generic[_T]):
     def __init__(self, file: str, loop: Any, executor: Any) -> None: ...
-    async def __aiter__(self) -> Iterator[str]: ...
-    async def __anext__(self) -> str: ...
+    async def __aiter__(self) -> Iterator[_T]: ...
```

Comment:
You could use this (`_C` needs to be defined):

```python
    async def __aiter__(self: Type[_C]) -> _C: ...
```

This way, the type returned by subclasses should be tighter. But please test this if you decide to go this way, as this is a fairly complex interaction.